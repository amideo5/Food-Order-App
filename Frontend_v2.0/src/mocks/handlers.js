// src/mocks/handlers.js
import { rest } from "msw";

export const handlers = [
  rest.post("http://localhost:8080/users/signin", (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.body("Successful")
    );
  }),

  rest.post("http://localhost:8080/users/createUser", (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.body("Registered")
    );
  }),

  rest.get("http://localhost:8080/users/name/*", async (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.body("Aryan")
    );
  }),

  rest.get("http://localhost:8080/orders/getOrder/*", (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.json({
        orderName: "Aryan",
        items: {
          orderItems: {
            rajma: "2",
            chicken: "3",
          },
        },
        cost: "2000",
      })
    );
  }),

  rest.put("http://localhost:8080/users/updateUser/*", (req, res, ctx) => {
    return res(ctx.status(200), ctx.body("User Updated"));
  }),
  
];
