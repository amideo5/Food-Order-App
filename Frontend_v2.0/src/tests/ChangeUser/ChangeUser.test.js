import React from "react";
import { render, screen } from "@testing-library/react";
import User from "../../Components/ChangeUser";
import userEvent from "@testing-library/user-event";
import { server } from "../../mocks/server";
import { rest } from "msw";
import { fireEvent, waitFor } from "@testing-library/react";

describe("User component", () => {
  test("Snapshot testing", () => {
    const { container } = render(<User />);

    expect(container).toMatchSnapshot();
  });

  it("updates the email input", () => {
    render(<User />);
    const emailInput = screen.getByPlaceholderText("Email");
    expect(emailInput).toBeInTheDocument();
    fireEvent.change(emailInput, "New Email Name");
    expect(emailInput).toHaveValue("");
  });

  it("updates the name input", () => {
    render(<User />);
    const nameInput = screen.getByPlaceholderText("Name");
    expect(nameInput).toBeInTheDocument();
    fireEvent.change(nameInput, "New Name");
    expect(nameInput).toHaveValue("");
  });

  it("updates the password input", () => {
    render(<User />);
    const passwordInput = screen.getByPlaceholderText("Password");
    expect(passwordInput).toBeInTheDocument();
    fireEvent.change(passwordInput, "12345678");
    expect(passwordInput).toHaveValue("");
  });

  it("updates the confirm password input", () => {
    render(<User />);
    const confirmpasswordInput =
      screen.getByPlaceholderText("Confirm Password");
    expect(confirmpasswordInput).toBeInTheDocument();
    fireEvent.change(confirmpasswordInput, "12345678");
    expect(confirmpasswordInput).toHaveValue("");
  });

  it("calls handleUpdate function when submit button is clicked", async () => {
    server.resetHandlers(
      rest.post("http://localhost:8080/users/updateUser/", (req, res, ctx) => {
        return res(
          // Respond with a 200 status code
          ctx.status(500),
          ctx.body("User Updated")
        );
      })
    );

    const props = {
      history: { push: jest.fn() },
    };

    const mockAlert = jest.fn();
    global.alert = mockAlert;

    const { getByPlaceholderText, getByText } = render(<User />);

    fireEvent.change(getByPlaceholderText("Email"), {
      target: { value: "test@test.com" },
    });
    fireEvent.change(getByPlaceholderText("Name"), {
      target: { value: "Aryan" },
    });
    fireEvent.change(getByPlaceholderText("Password"), {
      target: { value: "password" },
    });
    fireEvent.change(getByPlaceholderText("Confirm Password"), {
      target: { value: "password" },
    });
    const but = screen.getAllByText("Update");
    fireEvent.click(but[1]);

    await waitFor(() => expect(mockAlert).toBeCalled());
  });

  it("Normal flow", async () => {

    // server.resetHandlers(
    //   rest.put("http://localhost:8080/users/updateUser/", (req, res, ctx) => {
    //     return res(
    //       // Respond with a 200 status code
    //       ctx.status(200),
    //       ctx.body("User Updated")
    //     );
    //   })
    // );

    const props = {
      history: { push: jest.fn() },
    };

    const mockAlert = jest.fn();
    global.alert = mockAlert;

    const { getByPlaceholderText, getByText } = render(<User />);

    fireEvent.change(getByPlaceholderText("Email"), {
      target: { value: "test@test.com" },
    });
    fireEvent.change(getByPlaceholderText("Name"), {
      target: { value: "Aryan" },
    });
    fireEvent.change(getByPlaceholderText("Password"), {
      target: { value: "password" },
    });
    fireEvent.change(getByPlaceholderText("Confirm Password"), {
      target: { value: "password" },
    });
    const but = screen.getAllByText("Update");
    fireEvent.click(but[1]);

    await waitFor(() => expect(mockAlert).toBeCalledTimes(1));
    // expect(mockAlert).toBeCalledTimes(8)
  });

  it("All bad Both passwords are different", async () => {

    const props = {
      history: { push: jest.fn() },
    };

    const mockAlert = jest.fn();
    global.alert = mockAlert;

    const { getByPlaceholderText, getByText } = render(<User />);

    fireEvent.change(getByPlaceholderText("Email"), {
      target: { value: "test@test.com" },
    });
    fireEvent.change(getByPlaceholderText("Name"), {
      target: { value: "Aryan" },
    });
    fireEvent.change(getByPlaceholderText("Password"), {
      target: { value: "password" },
    });
    fireEvent.change(getByPlaceholderText("Confirm Password"), {
      target: { value: "password2" },
    });
    const but = screen.getAllByText("Update");
    fireEvent.click(but[1]);

    await waitFor(() => expect(mockAlert).toBeCalledTimes(1));
    // expect(mockAlert).toBeCalledTimes(8)
  });

});
