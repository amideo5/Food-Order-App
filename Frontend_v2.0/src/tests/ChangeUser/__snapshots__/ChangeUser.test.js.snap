// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component Snapshot testing 1`] = `
<div>
  <div
    class="page"
  >
    <nav
      class="m-0 py-3 navbar navbar-expand-lg navbar-dark bg-primary"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          NEX Food Ordering
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          />
          <a
            class="nav-link"
            href="/admin"
          >
            ADMIN
          </a>
          <a
            class="nav-link"
            href="/"
          >
            Sign In
          </a>
          <a
            class="nav-link"
            href="/signup"
          >
            Sign Up
          </a>
        </div>
      </div>
    </nav>
    <div
      class="form-container"
    >
      <div
        class="form-box-signup"
      >
        <div
          class="header-form"
        >
          <h4
            class="text-primary text-center"
          >
            <i
              class="fa fa-user-circle"
              style="font-size: 110px;"
            />
          </h4>
          <div
            class="image"
          />
        </div>
        <h2>
          Update
        </h2>
        <div
          class="body-form"
        >
          <form
            class="m-3"
          >
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                >
                  <i
                    class="fa fa-envelope fa-2x"
                  />
                </span>
              </div>
              <input
                class="form-control"
                name="email"
                pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,4}$"
                placeholder="Email"
                required=""
                title="Enter a valid Email"
                type="text"
                value=""
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                >
                  <i
                    class="fa fa-user fa-2x"
                  />
                </span>
              </div>
              <input
                class="form-control"
                name="name"
                placeholder="Name"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                >
                  <i
                    class="fa fa-lock fa-2x"
                  />
                </span>
              </div>
              <input
                class="form-control"
                name="password"
                pattern=".{8,}"
                placeholder="Password"
                required=""
                title="Eight or more characters"
                type="password"
                value=""
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                >
                  <i
                    class="fa fa-lock fa-2x"
                  />
                </span>
              </div>
              <input
                class="form-control"
                name="confirmPassword"
                pattern=".{8,}"
                placeholder="Confirm Password"
                required=""
                title="Eight or more characters"
                type="password"
                value=""
              />
            </div>
            <button
              class="btn btn-secondary btn-block"
              type="submit"
            >
              Update
            </button>
            <br />
            <br />
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
